### ISSUE TITLE
Test Version 2.5, error on login 

### ISSUE THREAD

**Issue by BurtGummer** [nextcloud/desktop#718]
Hello,

i install the newest file (Nextcloud-2.5.0.61459-daily-20181011.exe) from:
https://download.nextcloud.com/desktop/daily/Windows/?C=M;O=D

and install it on my Windows 10 Pro (1709). I start NextCloud Client, enter my URL https://example.org/cloud/ and then i got:

![image](https://user-images.githubusercontent.com/1249295/46866999-c8a16e00-ce23-11e8-8884-346108c79837.png)

I can not enter User/Password. Log from Nextcloud:

`
{"reqId":"ZPfg7XJhMrpUTuwcDuP6","level":3,"time":"2018-10-12T11:33:06+00:00","remoteAddr":"80.xxx.xxx.xxx","user":"--","app":"index","method":"GET","url":"\/cloud\/\/index.php\/login\/flow","message":{"Exception":"Exception","Message":"The requested uri(\/cloud\/index.php\/login\/flow) cannot be processed by the script '\/cloud\/\/index.php')","Code":0,"Trace":[{"file":"\/var\/www\/html\/nextcloud\/lib\/base.php","line":930,"function":"getRawPathInfo","class":"OC\\AppFramework\\Http\\Request","type":"->","args":[]},{"file":"\/var\/www\/html\/nextcloud\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/var\/www\/html\/nextcloud\/lib\/private\/AppFramework\/Http\/Request.php","Line":742,"CustomMessage":"--"},"userAgent":"Mozilla\/5.0 (Windows) mirall\/2.5.0daily (build 20181011) (Nextcloud)","version":"14.0.3.0"}`

The stable Clients (2.3.3.1) works well on all platforms. Any ideas?

**Comment by camilasan** [comment]
Hi, are you using a self signed certificate?

**Comment by BurtGummer** [comment]
Hi camilasan,

no, its a valid lets encrypt certificate.

**Comment by BurtGummer** [comment]
I have the same problem on a new, clean Windows 10.

**Comment by rullzer** [comment]
It is actually not the desktop client giving an error. it is your nextcloud server doing this.

**Comment by BurtGummer** [comment]
OK, please tell me if i can help something.

**Comment by MorrisJobke** [comment]
> @MorrisJobke do you maybe have any idea?

Never heard of this. I would also just search for it in the internet 🤷‍♂️

**Comment by MorrisJobke** [comment]
Is this then a webserver issue or one in the client? cc @rullzer

**Comment by BurtGummer** [comment]
Thank you all, without the / it works.

**Comment by quamis** [comment]
For me it let me login without using '/' but it failed to sync. It behaved like I didn't have any data on the server.

This is a client issue, not a server issue.

I've just had the same issue, with the 2.5 client (tested with appimage & deb). The appImage for 2.3 works as expected. 

The 2.5 app version seems to use some kind of webview to do the login, but the webview doesn't work as expected on hosts which don't have nextcloud installed in the root folder.

The 2.3 app version used a non-html based login, and probably din't  use the 'login/flow' files.

**Comment by quamis** [comment]
At this moment, the 2.3 client finished the sync, I've stopped it, and started 2.5, and everything seems ok. Seems like the conversion from 2.3 to 2.5 works ok, but adding a new account from scratch doesn't work on this configuration

**Comment by TeroPihlaja** [comment]
We are seeing this same issue with 2.5 client and 14.0.3 server. 2.3 client seems to work fine (with or without trailing slash).

Error in server log seems similar:
```
{"reqId":"I5SZUrbSUtLAqwwTlp13","level":3,"time":"2018-11-21T08:18:30+00:00","remoteAddr":"10.6.1.60","user":"--","app":"index","method":"GET","url":"\/owncloud\/\/index.php\/login\/flow","message":{"Exception":"Exception","Message":"The requested uri(\/owncloud\/index.p
hp\/login\/flow) cannot be processed by the script '\/owncloud\/\/index.php')","Code":0,"Trace":[{"file":"\/var\/www\/owncloud\/lib\/base.php","line":930,"function":"getRawPathInfo","class":"OC\\AppFramework\\Http\\Request","type":"->","args":[]},{"file":"\/var\/www\/own
cloud\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/var\/www\/owncloud\/lib\/private\/AppFramework\/Http\/Request.php","Line":742,"CustomMessage":"--"},"userAgent":"Mozilla\/5.0 (Windows) mirall\/2.5.0v2.5.0 (build 201811
12) (Nextcloud)","version":"14.0.3.0"}
```

**Comment by arnold-maderthaner** [comment]
have the same issue :(

**Comment by user23498723452** [comment]
I'm seeing the same erroe, and I do have an intentional certificate mismatch.  This is required due to the simple architecture of home-based networking.  I cannot implement a different (separate) certificate without requiring a proxy server - which is NOT realistic for a home-based implementation.

Testing the actual name for the cert proves the client can function.

What is the fix for a certificate that has some type of trust issue (like a selfsigned cert or direct IP access since there is no private/internal DNS)?

Is there a client option to ignore the cert trust issues?  The one raised seems to have no impact to the issue.

**Comment by Dias934** [comment]
> The error only happens when appending a `/`:
> 
> `https://example.org/cloud/` doesn't work
> `https://example.org/cloud` **does** work

Solved the issues! Thank you

**Comment by nielsd** [comment]
+1
same issue here with two user instances of NC from (at least) 13.x) up tp 14.06 (current stable). 
reported for current client on windows and osx. reinstallation and cleaning not working (but all of them had a owncloud client deinstalled before on that client machine if that helps in any way).

**Comment by muelli** [comment]
Has anyone filed a bug against the server side of things?
It seems silly for the server to crash with a 500.

**Comment by eighthave** [comment]
I just got this same error on nextcloud 15.0.4.0 running on Debian/stretch with nextcloud client 2.5.1git running on Ubuntu/LTS.  It was caused by the trailing slash on the URL while setting up the client.  Removing the trailing / fixes it for me.  Also, in the browser, both with or without the slash work fine.

This may seem trivial, but it took me a while to figure, including reading nextcloud debug logs.
